### PR TITLE
Refactor master-ip manifests

### PR DESCRIPTION
**Pull request by mborsz**
Refactoring the master-ip manifests to:
* share logic between kubemark and gce ('default')
* add support for multiple apiservers in gce (reusing the logic that already worked in kubemark)
/assign @mm4tt

### CODE REVIEW

**Comment by mm4tt** [comment]
/test pull-perf-tests-clusterloader2 
/test pull-perf-tests-clusterloader2-kubemark 

(To test perf-tests presubmits using boskos)

File: clusterloader2/go.mod [hunk 1]
```diff
@@ -0,0 +1,69 @@
+module k8s.io/perf-tests/clusterloader2
```
Comment (mm4tt) [inline clusterloader2/go.mod]: Remove this file?

**Comment by mm4tt** [comment]
As discussed personally, please make sure that no regression is introduced in GKE

**Comment by mm4tt** [comment]
An unintended side effect of this change is that our kubemark tests (e.g. presubmit) stopped scrapping etcd. Please fix

**Comment by mborsz** [comment]
See https://github.com/kubernetes/test-infra/pull/14911

File: clusterloader2/pkg/prometheus/manifests/default/kube-scheduler-service.yaml [hunk 2]
```diff
@@ -1,15 +0,0 @@
-apiVersion: v1
```
Comment (mm4tt) [inline clusterloader2/pkg/prometheus/manifests/default/kube-scheduler-service.yaml]: I think that this approach has the benefit of working both in GCE and GKE while the one from this PR won't work in GKE when it comes to system pods like kube-scheduler or kube-controller-manager, right?

**Comment by mm4tt** [comment]
/lgtm
/approve

Once this gets submitted could you check whether nothing got broken in our grafana dashboards? I guess some of the target names may have changed which may break some graphs.

**Pull request merged**